### PR TITLE
Frontend/requests

### LINKED ISSUE #42: Replace XML with fetch

### PR DESCRIPTION
**Pull request by aakhare**
**HTTP Request Functions**

- Sign Up Page is fully functional and connected to the server
   - On entering username and password, an associated account_id is created and stored into the database
   - Replaced XML and used fetch request with `createUser` function
      - Closes #42 
- Log In Page is fully functional and connected to the server 
   - On entering username and password, an associated account_id is returned from the server and stored into local storage 
   - Used fetch request with `getUser` function
- Task Page changes 
   - Used fetch request with `getTask` function
   - Used fetch request with `createTask` function 
       - On adding a task, an associated task_id is created and stored into the database
       - The user's account_id (obtained from local storage) is added to the task object on the frontend
   - Used fetch request with `deleteTask` function 
       - On hitting the delete button, the associated task is deleted from the database and local storage
   - Used fetch request with `updateTask` function 
      - On toggling the checkbox, the associated task is updated and changes are reflected accordingly in the database and local storage 

**Other Frontend Changes**

- Adjusted the sidebar, so correct icons are showing 
- Formatted the positioning of components on the task page

### CODE REVIEW

File: frontend/src/RequestOptions/user-requests.js [hunk 1]
```diff
@@ -1,54 +1,84 @@
...
         headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify({type:'create_user', user})
+        body: JSON.stringify({"type":'create_account',"params":user})
     };
-    fetch('https://localhost:8080/endpoint.php', requestOptions);
+    try {
+        console.log("hi");
+        console.log(requestOptions);
+        const response = await fetch('http://localhost:8000/endpoint.php', requestOptions);
+        console.log(response);
```
Comment (loshprung) [inline frontend/src/RequestOptions/user-requests.js]: Can some of these debug console.log's be removed?

Comment (aakhare) [reply frontend/src/RequestOptions/user-requests.js]: I removed the unnecessary ones, but I'm temporarily keeping the ones I still need to keep track of the requests and some local storage info

**Review by loshprung — APPROVED** [review]
Looks good

**Review by maddiewaldie — APPROVED** [review]
Looks good to me! :)

**Pull request closed without merging**